08-components-lab/src/04-multicolor.jsx

```diff
@@ -1,21 +1,43 @@
-/**
- * Lab 04:
- *
- * Implement a multi color picker component.
- *
- * Component should provide an <input> field to select a color,
- * and below 9 <div> elements.
- * Each <div> should have a slightly darker color than the previous one,
- * with the color selected by the user used as the middle color.
- *
- * See lab description on the website for a live demo.
- *
- * Hint: Use tinycolor library to perform color manipulations
- *
- */
-
-import tinycolor from 'tinycolor2';
-import React from 'react';
-
-export default React.createClass({
-});
+import React from "react";
+import tinycolor2 from "tinycolor2";
+
+export default class MultiColor extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {colors: []};
+    }
+
+    handleChange(e) {
+        var colors = [];
+        var selectedColor = e.target.value;
+        colors[5] = selectedColor;

```

Comment:
don't hardcode 5th place. It's better to pass color length as a prop from outside and put your color on half length